src/Features/Core/Portable/GenerateConstructorFromMembers/State.cs

```diff
@@ -59,7 +59,7 @@ private class State
 
                 this.Parameters = service.DetermineParameters(selectedMembers);
                 this.MatchingConstructor = service.GetMatchingConstructor(this.ContainingType, this.Parameters);
-                this.DelegatedConstructor = service.GetDelegatedConstructor(this.ContainingType, this.Parameters);
+                this.DelegatedConstructor = service.GetDelegatedConstructorBasedOnParameterTypes(this.ContainingType, this.Parameters);
```

Comment:
it's probably worth commentin the logic here.  i.e. why it's appropraite to be be matching based on parameter types and not names.  (and vice versa where you use hte BasedOnParameterNames helper).

Reply:
To be honest, I don't know exactly why the original author uses this strategy here. But I can put my understanding here.